tests/access_test.go

```diff
@@ -241,4 +245,80 @@ var _ = Describe("[rfe_id:500][crit:high][vendor:cnv-qe@redhat.com][level:compon
 			table.Entry("on vm restart", "virtualmachines", "restart"),
 		)
 	})
+
+	Describe("With regular OpenShift user", func() {
+		BeforeEach(func() {
+			tests.SkipIfNoCmd("oc")
+		})
+
+		const testUser = "testuser"
+
+		testRights := func(resource, right string) {
+			verbsList := []string{"get", "list", "watch", "delete", "create", "update", "patch", "deletecollection"}
+
+			for _, verb := range verbsList {
+				// AS A TEST USER
+				By(fmt.Sprintf("verifying user rights for verb %s", verb))
+				result, _, _ := tests.RunCommand(k8sClient, "auth", "can-i", "--as", testUser, verb, resource)
+				Expect(result).To(ContainSubstring(right))
+			}
+		}
+
+		Context("should fail without admin rights for the project", func() {
+			BeforeEach(func() {
+				By("Ensuring the cluster has new test user")
+				stdOut, stdErr, err := tests.RunCommandWithNS("", k8sClient, "create", "user", testUser)
+				Expect(err).ToNot(HaveOccurred(), "ERR: %s", stdOut+stdErr)
+
+				stdOut, stdErr, err = tests.RunCommandWithNS("", k8sClient, "project", tests.NamespaceTestDefault)
+				Expect(err).ToNot(HaveOccurred(), "ERR: %s", stdOut+stdErr)
+			})
+
+			AfterEach(func() {
+				stdOut, stdErr, err := tests.RunCommandWithNS("", k8sClient, "delete", "user", testUser)
+				Expect(err).ToNot(HaveOccurred(), "ERR: %s", stdOut+stdErr)
+			})
+
+			table.DescribeTable("should verify permissions on resources are correct for view, edit, and admin", func(resource string) {
+				testRights(resource, "no")
+			},
+				table.Entry("[test_id:]given a vmi", "virtualmachineinstances"),
```

Comment:
we need to add ids before merging, right?

Reply:
Yes. Working on it.

Reply:
@petrkotas 
Test cases are at the Jira card.
https://jira.coreos.com/browse/CNV-2370?focusedCommentId=116668&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-116668
Please update me (on the card) if something is missing